test/sagemaker_tests/tensorflow/inference/requirements.txt

```diff
@@ -9,3 +9,4 @@ fabric
 retrying
 gitpython
 toml
+packaging==21.3
```

Comment:
Does this fix the issue? I only see LegacyVersion being used in `test_utils/__init__.py`.

Reply:
it does as it's being used with efa (or connected) this resolves till we remove the logic. Note: PyTorch is in the same position with efa using the same process https://github.com/aws/deep-learning-containers/pull/2489/files#diff-aa45f4b64e705b76e10470cfe8ea6bbe8f77a67044011e274f404b1cde2fc37d

Reply:
I've tested that the new packaging.parse returns error on non-PEP 440 compatible string. 
`from packaging.version import parse
   ...: parse("1.1.1_x")` \

> InvalidVersion: Invalid version: '1.1.1_abc'
It might be better to just remove first condition here: https://github.com/aws/deep-learning-containers/blob/919779cf07aa8625ebb188f576042dfaf0e08298/test/test_utils/__init__.py#L412

pytorch_ver here would not be an incompatible version string.